x-pack/plugins/lens/server/migrations/common_migrations.ts

```diff
@@ -155,3 +161,40 @@ export const commonMakeReversePaletteAsCustom = (
   }
   return newAttributes;
 };
+
+export const commonRenameFilterReferences = (attributes: LensDocShape715<VisState716>) => {
+  const newAttributes = cloneDeep(attributes);
+  for (const filter of newAttributes.state.filters) {
+    filter.meta.index = filter.meta.indexRefName;
+    delete filter.meta.indexRefName;
+  }
+  return newAttributes;
+};
+
+const getApplyFilterMigrationToLens = (filterMigration: MigrateFunction<Filter>) => {
+  return (savedObject: { attributes: LensDocShape }) => {
+    return {
+      ...savedObject,
+      attributes: {
+        ...savedObject.attributes,
+        state: {
+          ...savedObject.attributes.state,
+          filters: savedObject.attributes.state.filters.map((filter) => filterMigration(filter)),
```

Comment:
@ppisljar my code here is assuming that the `Filter` migrations will accept a single `Filter` object. On the other hand, I just noticed that the [usage in the search source migrations](https://github.com/elastic/kibana/blob/d1774fe48081879c2377a52f6b6ba93f12a4b6c6/src/plugins/data/common/search/search_source/search_source_service.ts#L47-L52) implies that they will work on an array of `Filter` objects instead.

Just to be clear, is `Filter[]` or `Filter` the intended argument type for the future filter migrations?

Reply:
`Filter[]` is the correct type